### DISCUSSION TITLE
Dual authentication

### DISCUSSION THREAD

**Discussion by JackMai123** [immich-app/immich#6340] [Feature Request]
Hey team, what are your thoughts on beefing up the app’s security? I’m thinking of adding dual authentication – digital validation using QR codes or email.

**Comment by bo0tzz** [comment]
For advanced authentication approaches like 2FA, email validation, etc, you can use the OAuth feature.

**Reply by JackMai123** [reply]
Although the oauth is available, but it is hard to setup and the mobile app doesn't seem to work with Authelia. It would be nice if immich can add the 2 factor authentication built in.

**Reply by alextran1502** [reply]
We have other users that have successfully setup OAuth with authentik, authelia, google oauth as well. Maybe asking on Discord will help you with getting help on this end
